bears/general/InvalidLinkBear.py

```diff
@@ -40,18 +40,32 @@ def get_status_code(url, timeout):
             pass
 
     @staticmethod
+    def parse_pip_vcs_url(link):
+        splitted_at = link.split('@')[0]
+        splitted_tag = splitted_at.split('#')[0]
+        splitted_schema = splitted_tag[splitted_tag.index('+') + 1:]
+        return splitted_schema
+
+    @staticmethod
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'((git\+|bzr\+|svn\+|hg\+)?https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s('
+            r')%\'"`<>|\\]+|\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
```

Comment:
maybe instead of adding the `git+` etc. in front of the regex rather forbit an `@` in the later part of the regex to cut it off correctly? @jayvdb makes sense? Less code, then we could just change the regex and go with that